### PR TITLE
Implement  cfn-lint

### PR DESCRIPTION
**Pull request by avattathil**
Proposed Fix to issue https://github.com/aws-quickstart/taskcat/issues/107

The latest version of PyYAML broke the yaml validation. This PR uses cfn-lint for  yaml validation and supports 1.2 tags.

We can implement cfn-lint deeper in later commits (this PR just handles issue  https://github.com/aws-quickstart/taskcat/issues/107)

### CODE REVIEW

File: taskcat/stacker.py [hunk 1]
```diff
@@ -1693,6 +1699,32 @@ def check_yaml(self, yamlin, quite=None, strict=None):
             return False
         return True
 
+    def check_cfnyaml(self, yamlin, quite=None, strict=None):
```
Comment (jaymccon) [inline taskcat/stacker.py]: typo: quite == quiet ?

File: taskcat/stacker.py [hunk 2]
```diff
@@ -1034,7 +1036,7 @@ def generate_input_param_values(self, s_parms, region):
 
                 if licensecontent_re.search(param_value):
                     license_bucket = 'quickstart-ci-license'
-                    licensekey = (self.regxfind(licensecontent_re, param_value)).strip('/',1)[1]
+                    licensekey = '/'.join((self.regxfind(licensecontent_re, param_value)).split('/')[1:])
```
Comment (jaymccon) [inline taskcat/stacker.py]: should this commit be rolled into this pr ?

File: taskcat/stacker.py [hunk 3]
```diff
@@ -1693,6 +1699,32 @@ def check_yaml(self, yamlin, quite=None, strict=None):
...
+        try:
+            loader = cfnlint.decode.cfn_yaml.MarkedLoader(yamlin, None)
+            loader.add_multi_constructor('!', cfnlint.decode.cfn_yaml.multi_constructor)
+            if self.verbose:
+                if not quite:
+                    print(loader.get_single_data())
+        except Exception as e:
+            if strict:
+                print(E + str(e))
+                sys.exit(1)
```
Comment (sshvans) [inline taskcat/stacker.py]: Should we exit in one case and return false in another?

File: requirements.txt [hunk 4]
```diff
@@ -1,6 +1,60 @@
...
+aws-sam-translator==1.6.0
+backports.functools-lru-cache==1.5
+backports.ssl-match-hostname==3.5.0.1
+beautifulsoup4==4.6.0
+binaryornot==0.4.4
+boto3==1.7.20
+botocore==1.10.20
+certifi==2018.4.16
+cffi==1.11.5
+chardet==3.0.4
```
Comment (jaymccon) [inline requirements.txt]: shouldn't cfn-lint be in this list too ?

File: taskcat/stacker.py (hunk 2 above)
Comment (avattathil) [reply taskcat/stacker.py]: Its Andrews fix remove it from this Pr

File: requirements.txt (hunk 4 above)
Comment (avattathil) [reply requirements.txt]: Not sure why pip freeze does not pick it up I will add it manually

File: taskcat/stacker.py (hunk 3 above)
Comment (avattathil) [reply taskcat/stacker.py]: Can add similar to  this https://github.com/aws-quickstart/taskcat/blob/master/taskcat/stacker.py#L1616-L1619 once we get familiar with how syntax checks are

File: requirements.txt (hunk 4 above)
Comment (avattathil) [reply requirements.txt]: needed to use pip3, doh! (corrected)

File: taskcat/stacker.py (hunk 3 above)
Comment (sshvans) [reply taskcat/stacker.py]: 👍

File: taskcat/stacker.py (hunk 1 above)
Comment (avattathil) [reply taskcat/stacker.py]: lol yeah

**Pull request merged**